src/receive_imf.rs

```diff
@@ -696,6 +696,9 @@ async fn add_parts(
     prevent_rename: bool,
     verified_encryption: VerifiedEncryption,
 ) -> Result<ReceivedMsg> {
+    let is_bot = context.get_config_bool(Config::Bot).await?;
+    // Bots handle existing messages the same way as new ones.
+    let fetching_existing_messages = fetching_existing_messages && !is_bot;
```

Comment:
I think it's more straigtforward to reset `fetch_existing_messages` here for bots as it is checked in several places